src/ucp/api/ucp.h

```diff
@@ -3304,32 +3329,40 @@ ucp_atomic_fetch_nb(ucp_ep_h ep, ucp_atomic_fetch_op_t opcode,
 
 /**
  * @ingroup UCP_COMM
- * @brief Post an atomic fetch operation.
+ * @brief Post an atomic memory operation.
  *
- * This routine will post an atomic fetch operation to remote memory.
+ * This routine will post an atomic operation to remote memory.
  * The remote value is described by the combination of the remote
  * memory address @a remote_addr and the @ref ucp_rkey_h "remote memory handle"
- * @a rkey.
- * The routine is non-blocking and therefore returns immediately. However the
- * actual atomic operation may be delayed. The atomic operation is not considered complete
- * until the values in remote and local memory are completed.
- *
- * @note The user should not modify any part of the @a buffer or @a result after
- *       this operation is called, until the operation completes.
- * @note Only ucp_dt_make_config(4) and ucp_dt_make_contig(8) are supported in
- *       @a param->datatype, see @ref ucp_dt_make_contig
+ * @a rkey. The routine is non-blocking and therefore returns immediately.
+ * However the actual atomic operation may be delayed.
+ * If the reply buffer is not provided in @a param->reply_buffer, request handle
+ * is not returned and a user must call @ref ucp_ep_flush_nb or
+ * @ref ucp_worker_flush_nb to guarantee that the remote value has been updated.
+ * Otherwise, if fetch type of operation is being requested, the atomic
+ * operation is not considered complete until the values in remote and local
+ * memory are completed. The result of the operation will be stored in the
+ * @a param->reply_buffer memory buffer. In the case of CSWAP the value in
+ * @a param->reply_buffer will be swapped into the @a remote_addr if the
+ * condition is true.
+ *
+ * @note    The user should not modify any part of the @a buffer (or also
+ *          @a param->reply_buffer for fetch operations) after this operation is
```

Comment:
3350: seems a little redundant, what about just "operations), until this operation completes"?